rule/matcher_test.go

```diff
@@ -119,3 +120,87 @@ func TestMatcher(t *testing.T) {
 		})
 	}
 }
+
+var testRulesGlob = []Rule{
+	{
+		ID:             "foo1",
+		Match:          &Match{URL: "https://localhost:1234/<{foo*,bar*}>", Methods: []string{"POST"}},
+		Description:    "Create users rule",
+		Authorizer:     Handler{Handler: "allow", Config: []byte(`{"type":"any"}`)},
+		Authenticators: []Handler{{Handler: "anonymous", Config: []byte(`{"name":"anonymous1"}`)}},
+		Mutators:       []Handler{{Handler: "id_token", Config: []byte(`{"issuer":"anything"}`)}},
+		Upstream:       Upstream{URL: "http://localhost:1235/", StripPath: "/bar", PreserveHost: true},
+	},
+	{
+		ID:             "foo2",
+		Match:          &Match{URL: "https://localhost:34/<{baz*,bar*}>", Methods: []string{"GET"}},
+		Description:    "Get users rule",
+		Authorizer:     Handler{Handler: "deny", Config: []byte(`{"type":"any"}`)},
+		Authenticators: []Handler{{Handler: "oauth2_introspection", Config: []byte(`{"name":"anonymous1"}`)}},
+		Mutators:       []Handler{{Handler: "id_token", Config: []byte(`{"issuer":"anything"}`)}},
+		Upstream:       Upstream{URL: "http://localhost:333/", StripPath: "/foo", PreserveHost: false},
+	},
+	{
+		ID:             "foo3",
+		Match:          &Match{URL: "https://localhost:343/<{baz*,bar*}>", Methods: []string{"GET"}},
+		Description:    "Get users rule",
+		Authorizer:     Handler{Handler: "deny"},
+		Authenticators: []Handler{{Handler: "oauth2_introspection"}},
+		Mutators:       []Handler{{Handler: "id_token"}},
+		Upstream:       Upstream{URL: "http://localhost:3333/", StripPath: "/foo", PreserveHost: false},
+	},
+}
+
+func TestMatcherGlob(t *testing.T) {
+	type m interface {
+		Matcher
+		Repository
+	}
+
+	var testMatcher = func(t *testing.T, matcher Matcher, method string, url string, expectErr bool, expect *Rule) {
+		r, err := matcher.Match(context.Background(), method, mustParseURL(t, url))
+		if expectErr {
+			require.Error(t, err)
+		} else {
+			require.NoError(t, err)
+			assert.EqualValues(t, *expect, *r)
+		}
+	}
+
+	for name, matcher := range map[string]m{
+		"memory": NewRepositoryMemory(new(mockRepositoryRegistry)),
+	} {
+		t.Run(fmt.Sprintf("matcher=%s", name), func(t *testing.T) {
+			require.NoError(t, matcher.SetMatchingStrategy(context.Background(), configuration.Glob))
+			t.Run("case=empty", func(t *testing.T) {
+				testMatcher(t, matcher, "GET", "https://localhost:34/baz", true, nil)
```

Comment:
Not sure about merging this test, as we already have nested t.Run

Reply:
That's not a problem, we can nest `t.Run` I think merging the tests would save a bit of maintenance work later on :)